src/artifacts.js

```diff
@@ -176,7 +220,31 @@ api.declare({
   var isPublic = /^public\//.test(name);
   var details  = {};
   switch (storageType) {
+    case 'blob':
+      // Generate the details that we'll be using.
+      details.size = input.size;
+      details.sha256 = input.sha256;
+      if (input.parts) {
+        details.parts = input.parts;
+      }
+      // TODO: These ought to be configurable
+      details.provider = 's3'; // MAKECONFIG
+      details.region = 'us-east-1'; // MAKECONFIG
+      if (input.encoding) {
+        details.encoding = input.encoding;
+      }
+      if (isPublic) {
+        details.bucket = 'taskcluster-blobs'; // MAKECONFIG
+      } else {
+        details.bucket = 'taskcluster-private-blobs'; // MAKECONFIG
+      }
+
```

Comment:
Except that if we initiate the upload and the Entity creation fails, we have an upload floating around.  I thought about doing this, but I thought it was safer to not do it that way.  I know that the bucket policy should clean up those uploads, but it seems so easy to avoid that situation in the first place.